### ISSUE TITLE
Adding additional constraints to the VRP

### ISSUE THREAD

**Issue by jinkai1205** [reinterpretcat/vrp#103]
Is it possible to add additional constraints?

**Comment by reinterpretcat** [comment]
yes, that's possible when the logic is used as a library (rust crate). You would need to construct your own instance of VRP using desired constraints (wrapped by `Feature` concept). If you're using library through json pragmatic format, then it is not possible: this would require to support some domain specific language to express constraints in some form (e.g. AMPL).

**Comment by jinkai1205** [comment]
Is the solver able to handle tour length constraint？

**Comment by reinterpretcat** [comment]
yes, there is `maxDistance` and `maxDuration` constraints on vehicle's limits definition:

https://reinterpretcat.github.io/vrp/concepts/pragmatic/problem/vehicles.html

**Comment by jinkai1205** [comment]
Is there any documentation on how to add additional constraint? I'm not familiar with Rust crate.

**Comment by reinterpretcat** [comment]
No, there is no comprehensive documentation how to extend the solver code wise, only how to use it with json as a command line application
